arm-search/2015-08-19/swagger/search.json

```diff
@@ -0,0 +1,888 @@
+          "type": "string",
+          "readOnly": true,
+          "enum": [
+            "Invalid",
+            "AlreadyExists"
+          ],
+          "x-ms-enum": {
+            "name": "UnavailableNameReason",
+            "modelAsString": true
+          }
+        },
+        "message": {
+          "description": "A message that explains why the name is invalid and provides resource naming requirements. Available only if 'Invalid' is returned in the 'reason' property.",
+          "type": "string",
+          "readOnly": true
+        }
+      },
+      "description": "Output of check name availability API."
+    },
+    "AdminKeyResult": {
+      "properties": {
+        "primaryKey": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The primary admin API key of the Search service."
+        },
+        "secondaryKey": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The secondary admin API key of the Search service."
+        }
+      },
+      "description": "Response containing the primary and secondary admin API keys for a given Azure Search service."
+    },
+    "QueryKey": {
+      "properties": {
+        "name": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The name of the query API key; may be empty."
+        },
+        "key": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The value of the query API key."
+        }
+      },
+      "description": "Describes an API key for a given Azure Search service that has permissions for query operations only."
+    },
+    "ListQueryKeysResult": {
+      "properties": {
+        "value": {
+          "readOnly": true,
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/QueryKey"
+          },
+          "description": "The query keys for the Azure Search service."
+        }
+      },
+      "description": "Response containing the query API keys for a given Azure Search service.",
+      "x-ms-external": true
+    },
+    "Sku": {
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "The SKU of the Search service. Valid values include: 'free': Shared service. 'basic': Dedicated service with up to 3 replicas. 'standard': Dedicated service with up to 12 partitions and 12 replicas. 'standard2': Similar to standard, but with more capacity per search unit. 'standard3': Offers maximum capacity per search unit with up to 12 partitions and 12 replicas (or up to 3 partitions with more indexes if you also set the hostingMode property to 'highDensity').",
+          "enum": [
+            "free",
+            "basic",
+            "standard",
+            "standard2",
+            "standard3"
+          ],
+          "x-ms-enum": {
+            "name": "SkuName",
+            "modelAsString": false
```

Comment:
One note with `modelAsString: false` is if a new value is ever added, the client will be unable to handle it. When trying to deserialize the `name` property, it will be unable to map any values not listed here to a correct enum value (for languages that use real enums).